buildpacks/nodejs-npm-engine/src/main.rs

```diff
@@ -73,16 +73,27 @@ impl Buildpack for NpmEngineBuildpack {
         );
 
         let env = Env::from_current();
-        let inventory: Inventory =
-            toml::from_str(INVENTORY).map_err(NpmEngineBuildpackError::InventoryParse)?;
         let requested_npm_version =
             read_requested_npm_version(&context.app_dir.join("package.json"))?;
         let node_version = get_node_version(&env)?;
+        let existing_npm_version = get_npm_version(&env)?;
+
+        print::bullet("Determining npm package information");
+        let npm_package_packument =
+            resolve_requested_npm_packument(&context, &requested_npm_version)?;
 
         print::bullet("Installing npm");
-        let npm_release = resolve_requested_npm_version(&requested_npm_version, &inventory)?;
-        install_npm(&context, &npm_release, &node_version)?;
-        log_installed_npm_version(&env)?;
+        if existing_npm_version == npm_package_packument.version {
+            print::sub_bullet("Requested npm version is already installed");
```

Comment:
This output doesn't show up in any of the snapshots. Does it need to be exercised under test or is this branch not actually called? The `install_npm` also has cached logic. If possible I would prefer a single point of truth so we don't have to remember to update two locations if the logic changes.